doc/contributing/distribution.md

```diff
@@ -25,3 +25,11 @@ only as a semver-major change, unless the related feature or project is
 documented as experimental. In addition, Node.js includes external projects as
 internal dependencies. These may be replaced or removed at any time, provided
 that doing so is not a breaking change.
+
+## Placeholder executables
+
+Installing Node.js will not create "placeholder" executables: commands that
```

Comment:
this wording is pretty vague. `npx` can be considered to be a placeholder executable according to this definition or at least some interpretations of it

Reply:
I really don't see that, especially with the examples. The npx executable runs npx, which is included in our distribution. Npx itself isn't getting downloaded, it's downloading other things. Likewise for npm and Corepack.

Reply:
So to be clear, that means this policy would not block use from shipping a `corepack_yarn` executable, correct?

Reply:
It would block that. The policy says “refers to,” not “shares the exact same name as.” `corepack_yarn` refers to Yarn, just as `download_yarn` or `download_and_install_yarn` refer to Yarn.

Reply:
How does `npx` fits into this policy then? `npx yarn` downloads and execute Yarn, same as `corepack_yarn` would.

Reply:
For `npx yarn` the executable is `npx`, which doesn’t refer to Yarn. The text says “Installing Node.js will not create ‘placeholder’ executables;” Installing Node doesn’t create an `npx yarn` executable, it creates an `npx` executable. You may use that executable to run a command to download and install Yarn, and that’s fine, but the executable itself doesn’t refer to Yarn. Installing Node creates a `node` executable too and there’s some `node --eval` command that can download and install Yarn, but I’m not intending to ban the `node` executable either.

I understand you’re looking for loopholes, and perhaps the language can be improved, but the intent of the PR is very clearly laid out in the description. If you want to suggest alternative language that is clearer or stronger, that’s fine, but as I said in the TSC meeting that introduced this PR, the point is less about the specific language then the fact that reach an agreement about placeholder executables. If we can reach such a consensus, then we can move on to nail down the best possible language to express that.